### DISCUSSION TITLE
'AttributeContainer' is only available in iOS 15 or newer

### DISCUSSION THREAD

**Discussion by fury21** [LeoNatan/LNPopupController#556] [General] [answered]
I updated to the latest version 2.18.2 and now I get the error 'AttributeContainer' is only available in iOS 15 or newer when I try to configure appearance.titleTextAttributes.

Previously my code was like this:
let appearance = LNPopupBarAppearance()
         appearance.titleTextAttributes = [.font: UIFont(name: "Jost-SemiBold", size: 14)!,
                                           .foregroundColor: UIColor.black2_2E4756()]
                
and everything worked, but now I get the error Cannot assign value of type '[Any Hashtable : NSObject]' to type 'AttributeContainer'.

If I use the example code:
let appearance = LNPopupBarAppearance()
appearance.titleTextAttributes = AttributeContainer()
.font(UIFontMetrics(forTextStyle: .headline).scaledFont(for: UIFont(name: "Chalkduster", size: 14)!))
.foregroundColor(UIColor.yellow),

then I get the error 'AttributeContainer' is only available in iOS 15 or newer.

The current version of the project is ios13 and I don't want to upgrade it, how do I configure titleTextAttributes?

I'm using UIKit.

**Comment by LeoNatan** [comment, marked as answer]
Hello,
You can still use the old dictionary-based attributes for old operating systems like this:

```swift
if #available(iOS 15.0, *) {
	tabBarController?.popupBar.standardAppearance.titleTextAttributes = AttributeContainer()
		.font(UIFontMetrics(forTextStyle: .headline).scaledFont(for: UIFont(name: "Chalkduster", size: 14)!))
		.foregroundColor(UIColor.yellow)
} else {
	tabBarController?.popupBar.standardAppearance.__titleTextAttributes = [
		.font: UIFontMetrics(forTextStyle: .headline).scaledFont(for: UIFont(name: "Chalkduster", size: 14)!), 
		.foregroundColor: UIColor.yellow
	]
}
```

**Reply by LeoNatan** [reply]
You can also use `__titleTextAttributes` in iOS 15 and above as well, it's not really deprecated, and never will be, because in UIKit + Objective C, that's the only option available. It's just that the Swift refinement hides it, and there is no better way to expose it.
